logging/logging.go

```diff
@@ -803,7 +804,8 @@ func jsonValueToStructValue(v interface{}) *structpb.Value {
 		}
 		return &structpb.Value{Kind: &structpb.Value_ListValue{ListValue: &structpb.ListValue{Values: vals}}}
 	default:
-		panic(fmt.Sprintf("bad type %T for JSON value", v))
+		log.Printf("Error: bad type %T for JSON value", v)
```

Comment:
It looks like `toProtoStruct` returns an error. I think this method, and the method that calls it could be refactored to propagate an error returned here.

Reply:
Will do. More idiomatic for sure

Reply:
**Update**: On second thought, I don't think we should propagate this condition as an error. An invalid jsonValue shouldn't break `toLogEntry` entirely. Similarly, we do accept `nil` values (and just return a pb.NullValue). So in the case where `jsonValue` is of an unhandled type, we should return a `pb.NullValue` with a helpful stderr log for users.

However, if we absolutely want to error on this condition, then we should `log.Fatalf` at this step rather than propagate the error. Reasons (1) this is nested in a recursive function, error propagation will look ugly (2) we'll need to create a custom Error type, which adds complexity for a simple use case (3) the final stream downstream will exit anyway bc of this condition, so it might be better to exit program early rather than later.

Thoughts @codyoss ? I'll keep the former option for now

Reply:
It breaks it today, so it is no worse right? It would now just fail gracefully. As for `log.Fatalf`, that is basically a `log.Print` + `panic`. But your call on if you want to log, see other comment below.

Reply:
If you do choose to log in this library could you preface logs with "logging:". That way users know where they are coming from. This is a common idiom used with errors.

Reply:
Noted. Thanks for the pointer!

Maybe the right question I should ask is: should we be breaking on this error today? @codyoss 

I realize this scenario isn't strictly a user side error. It's happening when we fail to marshall a **valid JSON value** into a struct proto (somehow).

So in this case, maybe it's better to handle this similar to how we currently handle the `nil` json value? Which is to return a `structpb.Value_NullValue` and let the other logEntry fields be successfully logged (rather than error out the entire logEntry creation).

Reply:
Discussed with @simonz130 , we'll take the "help users log as much as can be logged"  approach while logging to stderr the details of what failed to marshal. Using the log preface Cody suggested.

Thanks all!